src/main/java/mysticalmechanics/util/Misc.java

```diff
@@ -8,24 +9,30 @@
 import net.minecraft.world.WorldServer;
 
 public class Misc {
-    static double epsilion = 0.00001;
+	static double epsilion = 0.00001;
 
-    public static boolean isRoughlyEqual(double a, double b) {
-        return a == b || Math.abs(a - b) < epsilion;
-    }
+	public static boolean isRoughlyEqual(double a, double b) {
+		return a == b || Math.abs(a - b) < epsilion;
+	}
 
-    public static void syncTE(TileEntity tile) {
-        World world = tile.getWorld();
-        if(world instanceof WorldServer) {
-            SPacketUpdateTileEntity packet = tile.getUpdatePacket();
-            if (packet != null) {
-                PlayerChunkMap chunkMap = ((WorldServer) world).getPlayerChunkMap();
-                int i = tile.getPos().getX() >> 4;
-                int j = tile.getPos().getZ() >> 4;
-                PlayerChunkMapEntry entry = chunkMap.getEntry(i, j);
-                if(entry != null)
-                    entry.sendPacket(packet);
-            }
-        }
-    }
+	public static void syncTE(TileEntity tile, boolean broken) {
+		World world = tile.getWorld();		
+		if (world instanceof WorldServer) {
+			PlayerChunkMap chunkMap = ((WorldServer) world).getPlayerChunkMap();
+			SPacketUpdateTileEntity packet = tile.getUpdatePacket();
+			if (packet != null) {
+				int i = tile.getPos().getX() >> 4;
+				int j = tile.getPos().getZ() >> 4;
+				PlayerChunkMapEntry entry = chunkMap.getEntry(i, j);				
+				if (entry != null) {
+					if(broken){
```

Comment:
so if the TE packet causes the problem, could we just not send it if the tile is broken to fix the problem without the extra blockchange packet?

Reply:
can try and test it im not sure if your going to get things like power change though, as far as i can see the server is sending the last packet for that tile entity to the client before the client realizes that the block is actually missing.

Reply:
ok is this otherwise ready for merge?

Reply:
ya sorry i missed that comment, unless you want further testing.